### PR TITLE
feat: integrate stylelint to css-lint rule

### PR DESCRIPTION
**Pull request by ling1726**
Follow up of #430 and integrates stylelint to lint atomic css rules in eslint.

Uses [synckit](https://github.com/un-ts/synckit) to make the styleint calls synchronous. The method behind synckit [is explained a bit here](https://samthor.au/2021/block-nodejs-main-thread/)

Exports the `css-lint` rule as unstable for now

### CODE REVIEW

File: packages/eslint-plugin/package.json [hunk 1]
```diff
@@ -12,9 +12,11 @@
     "@babel/core": "^7.12.13",
     "@typescript-eslint/utils": "^5.47.0",
     "csstype": "^3.1.2",
+    "synckit": "^0.8.5",
     "tslib": "^2.1.0"
   },
   "peerDependencies": {
-    "eslint": "^7.5.0 || ^8.0.0"
+    "eslint": "^7.5.0 || ^8.0.0",
+    "stylelint": "15.x"
```
Comment (layershifter) [inline packages/eslint-plugin/package.json:20]: This probably should be a direct dependency, actually.

File: packages/eslint-plugin/src/rules/css-lint/.stylelintrc.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/**
+ * A test stylelint config for tests
+ */
+
+module.exports = {
+  rules: {
+    'selector-anb-no-unmatchable': [true],
```
Comment (layershifter) [inline packages/eslint-plugin/src/rules/css-lint/.stylelintrc.js:7]: "anb"?

File: packages/eslint-plugin/src/rules/css-lint/.stylelintrc.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/**
```
Comment (layershifter) [inline packages/eslint-plugin/src/rules/css-lint/.stylelintrc.js:1]: Let's name this file directly to include `-test`. Also please check that it does not leak to `dist` folder.

File: packages/eslint-plugin/src/rules/css-lint/.stylelintrc.js (hunk 2 above)
Comment (layershifter) [reply packages/eslint-plugin/src/rules/css-lint/.stylelintrc.js:7]: Edit: i see, it's a valid rule.

**Review by layershifter — APPROVED** [review]

File: packages/eslint-plugin/package.json (hunk 1 above)
Comment (ling1726) [reply packages/eslint-plugin/package.json:20]: well it's probably never going to be put into a bundle, so I don't mind

**Pull request merged**